source/common/io/io_uring_worker_impl.h

```diff
@@ -235,25 +235,29 @@ class IoUringServerSocket : public IoUringSocketEntry {
   // handler.
   Request* read_req_{};
   // TODO (soulxu): Add water mark here.
-  Buffer::OwnedImpl buf_;
+  Buffer::OwnedImpl read_buf_;
   // TODO (soulxu): using queue for completion.
   absl::optional<int32_t> read_error_;
 
-  // For write.
+  // For write. io_uring socket will write sequentially in the order of write_buf_ and shutdown_
+  // Unless the write_buf_ is empty, the shutdown operation will not be performed.
   Buffer::OwnedImpl write_buf_;
-  struct iovec* iovecs_{nullptr};
-  Request* write_req_{nullptr};
+  // shutdown_ has 3 states. A absl::nullopt indicates the socket has not been shutdown, a false
+  // value represents the socket wants to be shutdown but the shutdown has not been performed or
+  // completed, and a true value means the socket has been shutdown.
+  absl::optional<bool> shutdown_{};
```

Comment:
This is hidden multiple statuses in this single variable, but it is enough for now.